### ISSUE TITLE
[OS X] Atom shell command should not show a (temporary) second dock icon when Atom is already open

### ISSUE THREAD

**Issue by rsanchezsaez** [atom/atom#2964]
Having _Atom_ running, when you open additional files (or try to open a file that you have already open) from the _Terminal_, _OS X's Dock_ will show a temporary second _Atom_ icon (and then it goes away). This should not be the case.

See how _TextMate_ 1.5.x doesn't do this when you use the _mate_ shell command. Same for opening _Sublime Text_ running _subl_ from the terminal.

**Comment by mnquintana** [comment]
Correct me if I'm wrong, but it seems like this has been resolved already? I haven't experienced this on any version of OS X since sometime last year.

**Comment by postcasio** [comment]
No, this still happens, but it's very quick. The shell script always opens a new instance of Atom, and Atom itself checks to see if there's another copy running as part of its startup - but at this point Atom Shell has already created the Dock icon.

**Comment by philipgiuliani** [comment]
Yes i can confirm thats still present.

dominic <notifications@github.com> schrieb am Mo., 16. März 2015 18:09:

**Comment by lhoworko** [comment]
Confirmed still present in 0.188

**Comment by CvX** [comment]
This issue, while a bit stale is as relevant and valid as ever. I'd ask for reopening.

**Comment by stackptr** [comment]
Agreed, I still run into this issue, though apparently less often these days

**Comment by rsanchezsaez** [comment]
This is still reproducible on Atom 1.28.0 running on macOS 10.13.5 (17F77) using the steps described in the original issue.

**Comment by jerep** [comment]
This is more annoying now in macOS Mojave with the "Show recent applications in Dock" setting on. Temporary dock icons end up to the recent applications:

<img width="262" alt="screenshot 2018-09-26 at 17 37 52" src="https://user-images.githubusercontent.com/1065535/46087523-2fcbeb00-c1b3-11e8-929a-b373f8166ca2.png">

**Comment by ber4444** [comment]
Just a note that this is not only from the Terminal. Same issue happens when using Atom as a conflict resolver in Github Desktop. My recent apps menu is taken over now by multiple Atom icons.

**Comment by sadick254** [comment]
This seems to be an issue with how macOS opens applications.

**Current state**
The second instance is visible for a few seconds and then disappears.

![where we are at](https://user-images.githubusercontent.com/5238135/91180169-1e1d5600-e6f0-11ea-884b-f6c4e5ab24d4.gif)

**Best state we can get to**
The second instance is not visible at all, but visually looks like macOS is trying to display the app icon.

![Best we can get to](https://user-images.githubusercontent.com/5238135/91180228-2d9c9f00-e6f0-11ea-966f-3d35738b0123.gif)

**Comment by sadick254** [comment]
```
OS: macOS Catalina 10.15.6
```
The gif that I have posted above is from my experimentation and not the actual atom application.
